### ISSUE TITLE
Allow users to merge comics.

### ISSUE THREAD

**Issue by bareheiny** [comixed/comixed#76]
As a user, I would like to be able to merge multiple comics to create a single entry in the CX DB (and the associated cbz / cbr file if possible).

As I have been cleaning my library, it has occurred to me that I may want to merge some individual comic files into a single consolidated one.

As this would create a comic that may not exist in ComicVine, I would like to have CX merge any associated metadata - such as synopsis and creative team.  If possible, I would also like CX to actually merge the relevant cbz / cbr files as well - otherwise I'm thinking there would be a problem with the file hash.

A use case would be the three issue Spawn: The Impaler series.  As I would ideally read it in one go, I would like to merge it into a single file called "Spawn: The Impaler", however if the creative team or characters changed through the issues, I would need the metadata for each issue linked to the new combined one.

**Comment by mcpierce** [comment]
This is an interesting feature, and brings to mind for me the idea of evolving the ComicInfo.xml file. I've never been able to find a conclusive source for where that file was first defined, but I think it's growing a bit dated now. And looking at merging multiple comics together might be an excuse to come up with a more updated standard, one that is able to refer to a combined comic like this (lets call it a synthetic trade) and put into it the metadata for the individual issues.

**Comment by bareheiny** [comment]
An updated standard would possibly need buy-in from the other main library and reader software developers I'd think.  Wouldn't do much for attracting new CX users if they found the info file couldn't be used in their chosen readers.

For now, I'm thinking something simple and less elegant :)

Once the comics are in CX and have been scrapped...a function that will simply change the comic id key's in the database so all the metadata is coded to a single issue, and (ideally) the comic files are combined.

Could be an issue with page names and numbering here though.  I may need to think about this a bit more.

**Comment by mcpierce** [comment]
True, that's the one thing that's held me back from starting on one is getting some buy in.

**Comment by bareheiny** [comment]
So you can merge files via the CR export option.

I didn't play with it too much, but it merged fine.  From what I saw, they comics were merged in the order I'd selected them, every page was renumbered (starting with 1), and the covers were given a slightly more informative name - such as "P0001 - Area 52 V2001 #1 (2001_1.jpg" and "P0029 - Area 52 V2001 #2 (2001_3.jpg"

It looks like the ComicInfo.xml is taken from the first issue in the list, so there's a loss of some metadata and the summary / plot would be specific to the first issue.

Overall, it looked pretty comprehensive - so it may be worth having a look at it to feed into a potential CX implementation.

**Comment by bareheiny** [comment]
Better option (in my view) - simply unzip the comics to be merged, then zip the entire set of folders.

This saves having to rename each page, and also would make it easier to un-merge them if the user wanted to.

I tested this with one comic reader (Cover), and it reads the comics fine...and in the correct order.  I do need to play around with the info files to see what happens when there are multiples (one in each folder)...and I think a info file should be created in the base folder as well.
